processes/WIP-bug_bounty_proposal.md

```diff
@@ -0,0 +1,73 @@
+
+A bug bounty program is a mechanism to formally engage security researchers and developers in a way that provides benefits to the community along multiple vectors:
+
+1. Improved Security Depth of Coverage • Finding security defects is specialized work that benefits from deep expertise and fresh eyes
+1. Validate Nodejs as a secure platform • Expanding the community into the security world is a growth area and enables new audiences for nodejs to be proposed
+1. Improved Security Scope of Coverage • Reduced Risk we have bugs that are undiscovered currently exposing ourselves and our users to unnecessary risk
+1. Improved Process in SDLC • Ongoing analysis of discoveries provides an opportunity to improve security through practices and/or security focus and tooling.
+
+## Proposal Process
+
+The proposed process for considering a Bug Bounty program makes heavy reference to existing work that the Security WG has already started (establishing a triage team, organized disclosure system, etc.) and ongoing partnership with HackerOne. High level process:
+
+-[ ] Security WG reviews readiness and desire to have such a program (presume yes unless someone objects)
+-[ ] Security WG presents proposal to TSC (see next steps on how to get there)
+-[ ] TSC / Security WG bring funding proposal before Board
+-[ ] Security WG & TSC establish Bug Bounty Leader & Bug Bounty Team (will take additional coordination across org)
+
+## Next steps
+
+1. Core team readiness review and architectural decisions
+
+   - goals/kpi (scope, responsiveness goals, impact of defects, team size, budget)
+   - private vs. public
+   - existing security analysis
+   - backlog validation
+
+1. Engage other committies to let them know of the proposal (community committee as there is a PR element, others?)
+1. Review and evolve draft note into a proposal
+1. Create draft Security Page updates that describe the rules of the road
+
+## Readiness review and questions for input
+
+As preparation, the Security WG already has an organization within HackerOne and the general framework as well as the technical tools to offer a Bug Bounty program.
+
+1. Do we have a security assessment of the current node.js core? *no formal assessment, should not block, but increases risk of public bounty program*
+1. Do we have any scope limitations on what we would declare? *yes, we do, just what are they exactly?*
+1. Should we bounty "LTS" version bugs and non-LTS the same *(gut is yes)*? Release vs. HEAD *(gut is no)*?
+1. Do we have enough volunteer staff (need to get HackerOne help to know what the right number might should be)
```

Comment:
Depends on how many reports you're getting. Just want to make sure you have enough resources to reply to people in a reasonable time period and follow-up as needed. Would help to know how many reports security@ is getting on a regular basis right now.

Reply:
I need someone that responds to them to address the current inbound rate. My assumption is that if we make an offer of payment it would increase the figure. Both for legit reports as well as probably out of scope or otherwise invalid reports. I only see them after disclosed though so I've got no idea what the numbers are right now.

Reply:
You’re entirely correct. Adding a bounty will increase the number of received reports, both good and bad ones. This is especially true right at the beginning after any sort of announcement. That’s always one of the biggest increase times. So, yes, having some data would greatly help here.